universe/System.cpp

```diff
@@ -310,6 +310,32 @@ bool System::Contains(int object_id) const {
     return m_objects.find(object_id) != m_objects.end();
 }
 
+bool System::ContainsHostileFleet(int empire_id) const {
+    for (auto obj_id : m_objects) {
+        auto object = GetUniverseObject(obj_id);
+        if (!object) {
+            ErrorLogger() << "Invalid object " << obj_id;
+            continue;
+        }
+
+        auto obj_type = object->ObjectType();
+        if (obj_type <= INVALID_UNIVERSE_OBJECT_TYPE || obj_type >= NUM_OBJ_TYPES) {
+            ErrorLogger() << "Invalid type for object " << obj_id;
+            continue;
+        } else if (obj_type != OBJ_FLEET) {
+            continue;
+        }
+
+        if (object->Owner() != ALL_EMPIRES &&
+            Empires().GetDiplomaticStatus(empire_id, object->Owner()) != DIPLO_WAR)
```

Comment:
Why not allow `ALL_EMPIRES`?  They are at war with everybody, but it is still a valid state.

I think this will also fix the issue @geoffthemedio observed with the monsters.

Reply:
No, I don't think so, for two reasons-- 1) this code already works to treat any ```ALL_EMPIRES``` fleet as at-war, and essentially how all similar code works because (2) GetDiplomaticStatus() returns  INVALID_DIPLOMATIC_STATUS if either specified empire is ALL_EMPIRES.  (I can only guess that the design decision there was to emphasize that everything ALL_EMPIRES is always at war with everyone else.

Reply:
if fleet->Owner() == ALL_EMPIRES, the function proceeds to return True(has a hostile ship), instead of checking for diplo status and potentially continuing to next fleet.
I styled a negate here, so that an additional check could easily be added if later desired, possibly for Fleet::HasArmedShips/HasFighterShips.

Reply:
@Dilvish-fo Opps, I did not see your comments prior to posting mine. :/